perfkitbenchmarker/benchmarks/fio_benchmark.py

```diff
@@ -77,14 +77,16 @@
 flags.DEFINE_boolean('against_device', False,
                      'Unmount the device\'s filesystem so we can test against '
                      'the raw block device. If --generate-scenarios is given, '
-                     'will generate a job file that uses the block device.')
+                     'will generate a job file that uses the block device. '
+                     'When using --against_device, default to pre-filling the '
+                     'device to 100% unless modified with --device_fill_size.')
 flags.DEFINE_string('device_fill_size', '100%',
```

Comment:
A new flag Sounds good to me.

On Wednesday, October 7, 2015, Anthony Voellm aka Tony the P3rfguy <
notifications@github.com> wrote:

> In perfkitbenchmarker/benchmarks/fio_benchmark.py
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/515#discussion_r41439724>
> :
>
> > @@ -77,14 +77,16 @@
> >  flags.DEFINE_boolean('against_device', False,
> >                       'Unmount the device\'s filesystem so we can test against '
> >                       'the raw block device. If --generate-scenarios is given, '
> > -                     'will generate a job file that uses the block device.')
> > +                     'will generate a job file that uses the block device. '
> > +                     'When using --against_device, default to pre-filling the '
> > +                     'device to 100% unless modified with --device_fill_size.')
> >  flags.DEFINE_string('device_fill_size', '100%',
>
> > After some discussion, we figured out a smart way to handle this: we
> will simply check for the presence of the flag in the
> test-with-file-system mode case, if the flag is present, then we will
> follow user's order, otherwise, we won't fill the device. In the
> test-against-device mode, we will just use the flag as is with the default
> value being 100%. (so if the flag is not present in the test-against-device
> mode, the default value of 100% will apply). This is how I read the code.
> However its not clear. Perhaps there needs to be a third flag.
> against_device device_fill_size prefill_device_when_using_filesystem By
> adding the third flag you make it clear the behavior without having to read
> the code and deduce special settings. Please add the additional flag so
> this is clear in the code.
> … <#15043eb94aba3c4a_>
> On Wed, Oct 7, 2015 at 1:00 PM, stfeng <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote: In
> perfkitbenchmarker/benchmarks/fio_benchmark.py <
> https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/515#discussion_r41438756>
> : > @@ -77,14 +77,16 @@ > flags.DEFINE_boolean('against_device', False, >
> 'Unmount the device\'s filesystem so we can test against ' > 'the raw block
> device. If --generate-scenarios is given, ' > - 'will generate a job file
> that uses the block device.') > + 'will generate a job file that uses the
> block device. ' > + 'When using --against_device, default to pre-filling
> the ' > + 'device to 100% unless modified with --device_fill_size.') >
> flags.DEFINE_string('device_fill_size', '100%', Tony, actually it's tricker
> than it appears... What we want is: 1) if the user is testing against the
> device directly, the default value of --device_fill_size should be 100%
> because in this mode, the user should fill the device entirely before the
> fio run. 2) If the user is testing using a file system, then the default
> behavior should be no-fill (0%) because that is unnecessary as fio will
> create the test file first and use that file for testing. However, on some
> providers like AWS, they require pre-warming of the device no matter what,
> so we still need to provide the user a choice to specify --device_fill_size
> to be 100% when they run fio on those providers in file system mode. To sum
> it up: the desired default value of this flag changes as the scenario
> changes. After some discussion, we figured out a smart way to handle this:
> we will simply check for the presence of the flag in the
> test-with-file-system mode case, if the flag is present, then we will
> follow user's order, otherwise, we won't fill the device. In the
> test-against-device mode, we will just use the flag as is with the default
> value being 100%. (so if the flag is not present in the test-against-device
> mode, the default value of 100% will apply). — Reply to this email directly
> or view it on GitHub <
> https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/515/files#r41438756>
> .
> -- Anthony F. Voellm (aka Tony) Google Voice: (650) 516-7382 <
> https://www.google.com/voice/b/0?pli=1#phones> Blog:
> http://perfguy.blogspot.com/ Benchmark like a pro... PerfKitBenchmarker <
> https://github.com/GoogleCloudPlatform/PerfKitBenchmarker>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/515/files#r41439724>
> .
>